src/snowflake/connector/telemetry_oob.py

```diff
@@ -154,7 +155,7 @@ def __init__(self):
         self.num_of_retry_to_trigger_telemetry = DEFAULT_NUM_OF_RETRY_TO_TRIGGER_TELEMETRY
         self.context = dict()
         self.connection_params = dict()
-        self.deployment = None
+        self.deployment = TelemetryServerDeployments.PROD
```

Comment:
__instance does get assigned in the init function. This is necessary because we may be sending OOB telemetry before we have the connection parameters and call update_context on this class